src/main/kotlin/io/github/anki/anki/service/CardsService.kt

```diff
@@ -1,32 +1,74 @@
 package io.github.anki.anki.service
 
+import io.github.anki.anki.controller.exceptions.CardDoesNotExistException
 import io.github.anki.anki.repository.mongodb.CardRepository
+import io.github.anki.anki.repository.mongodb.document.MongoCard
 import io.github.anki.anki.service.model.Card
 import io.github.anki.anki.service.model.mapper.toCard
 import io.github.anki.anki.service.model.mapper.toMongo
+import org.bson.types.ObjectId
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 
 @Service
 class CardsService(
     private val cardRepository: CardRepository,
+    private val deckService: DeckService,
 ) {
-
-    fun createNewCard(card: Card): Card {
+    fun createNewCard(userId: String, card: Card): Card {
+        deckService.getDeckByIdAndUserId(card.deckId, userId)
         LOG.info("Creating new card: {}", card)
-        return cardRepository.insert(
-            card.toMongo(),
-        )
+        return cardRepository.insert(card.toMongo())
             .toCard()
             .also { LOG.info("Successfully saved new card: {}", it) }
     }
 
-    fun deleteCard(cardId: String) {
+    fun getAllCardsFromDeck(deckId: String, userId: String): List<Card> {
+        deckService.getDeckByIdAndUserId(deckId, userId)
+        LOG.info("Getting MongoCards by deckId = {}", deckId)
+        return cardRepository.findByDeckId(ObjectId(deckId))
+            .map { it.toCard() }
+            .also { LOG.info("Got MongoCards from DB {}", it) }
```

Comment:
```suggestion
        LOG.info("Finding cards by deckId = {}", deckId)
        return cardRepository.findByDeckId(ObjectId(deckId))
            .map { it.toCard() }
            .also { LOG.info("Found cards: {} by deckId={}", it, deckId) }
```

Reply:
And user will get empty list but not message that he has no such deck
In Ajax if i want to get list of hub in company when i am not employee what will i get? i will get permission denied